agent/dns.go

```diff
@@ -206,6 +207,29 @@ func (d *DNSServer) handlePtr(resp dns.ResponseWriter, req *dns.Msg) {
 				m.Answer = append(m.Answer, ptr)
 				break
 			}
+
+			sargs := structs.NodeSpecificRequest{
```

Comment:
I don't get what you are doing there... does the `Catalog.NodeServices` should not be when the node has been found instead ?

In my understanding, you are creating LOTS of requests to the Consul server for nothing

I don't get why you created the label Outer neither...

Reply:
The code i'madding is to find the service(name) of an ip address, eg 2001:db8:85a3::1, so I have to check the ip address of all the registered services using Catalog.NodeServices. 

If we find the address, we create the response and break out of the 2nd level for loop using the Outer label.